notebooks/ullyses-walkthrough.ipynb

```diff
@@ -311,58 +392,228 @@
+    "    prevspec_filename = glob.glob(f'*{ull_targ}*_preview-spec.fits')[0]\n",
+    "    spec_file = Path(prevspec_filename)\n",
+    "\n",
+    "    # Make an array to hold the level 3 filenames\n",
+    "    level3 = glob.glob(f'*{ull_targ}*aspec*')\n",
+    "    \n",
+    "    # Set up the plot - start with plotting the level 4 products\n",
+    "    fig = plt.figure()\n",
+    "    gs = fig.add_gridspec(3, hspace=0)\n",
+    "    axs = gs.subplots(sharex=True, sharey=False)\n",
+    "\n",
+    "    # preview-spec data\n",
+    "    wavelength, flux, error, snr, gratings = get_non_variable_data(spec_file)  \n",
+    "    \n",
+    "    axs[0].plot(wavelength, flux, color='k', label='preview-spec')\n",
+    "    axs[1].plot(wavelength, error, color='k')\n",
+    "    axs[2].plot(wavelength, snr, color='k')\n",
+    "    \n",
+    "    # Then use a for loop to add the level 3 products on top\n",
+    "    for prod in level3:\n",
+    "        # level 3 product data\n",
+    "        prod_wavelength, prod_flux, prod_error, prod_snr, prod_gratings = get_non_variable_data(prod) \n",
+    "\n",
+    "        axs[0].plot(prod_wavelength, prod_flux, alpha=0.8, label=prod_gratings)\n",
+    "        axs[1].plot(prod_wavelength, prod_error, alpha=0.8)\n",
+    "        axs[2].plot(prod_wavelength, prod_snr, alpha=0.8) \n",
+    "        \n",
+    "    \n",
+    "    # Add all the labels & limits\n",
+    "    plt.suptitle(f'{targ.upper()} aspec')\n",
+    "    axs[0].set_ylabel('Flux\\n[ergs/s/cm^2/$\\mathrm{\\AA}$]')\n",
+    "    axs[1].set_ylabel('Error\\n[ergs/s/cm^2/$\\mathrm{\\AA}$]')\n",
+    "    axs[2].set_ylabel('SNR')\n",
+    "    plt.xlabel('Wavelength [$\\mathrm{\\AA}$]')\n",
+    "    axs[0].legend(ncol=len(level3)+1, loc='upper left')\n",
+    "\n",
+    "    axs[0].set_xlim(wavelength.min(), wavelength.max())\n",
+    "    if ull_targ == 'v-xx-cha':\n",
+    "        axs[0].set_ylim(-1E-15, 5E-14)\n",
+    "        axs[1].set_ylim(-1E-15, 5E-15)\n",
+    "    else:\n",
+    "        # set the limit based on the preview-spec to avoid grating edges\n",
+    "        axs[0].set_ylim(np.nanmin(flux), np.nanmax(flux)) \n",
+    "        axs[1].set_ylim(np.nanmin(error), np.nanmax(error)) \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Non-Variable Provenance Table\n",
     "\n",
-    "Select metadata for each spectrum that contributes to the combined spectrum in the SCIENCE extension will populate a row in the provenance table. The fields in the table are metadata harvested from the headers of the contributing spectra."
+    "Select metadata for each spectrum that contributes to the combined spectrum in the SCIENCE extension will populate a row in the provenance table. The fields in the table are metadata harvested from the headers of the contributing spectra. Notice that we already used the \"DISPERSER\" column in the plot labels above to see all contributing gratings in that product."
```

Comment:
May want to reword the first sentence a bit. Maybe something like: "Select metadata for each spectrum that contributes to the combined spectrum in the SCIENCE extension will populate a row in the provenance table." -> "For each spectrum that contributes to a given HLSP, select metadata is available in the provenance table."